Add staking section to unified SDK (#1050)
casper-network/docs: source/docs/casper/developers/dapps/sdk/client-library-usage.md

</Tabs>

Once submitted, the above snippet will print the deploy hash in the console.

## Staking

Token staking is a fundamental aspect of the Casper Network, whereby users lock up tokens as collateral in exchange for the ability to participate in the blockchain's consensus mechanism and earn rewards. This delegated Proof-of-Stake consensus mechanism is crucial for the network's effective operation. With the aid of any of the Casper SDKs, you can delegate your tokens to validators and participate in staking on the network.

The delegation functionality is available as a smart contract, which can be found in the [casper-node](https://github.com/casper-network/casper-node) repository. To delegate tokens, first clone the repository:

```bash
git clone https://github.com/casper-network/casper-node.git
cd casper-node/
```

Then compile the [delegate contract](https://github.com/casper-network/casper-node/blob/dev/smart_contracts/contracts/client/delegate/src/main.rs):

```bash
make setup-rs
make build-contract-rs/delegate
```

> Now, assuming that you cloned `casper-node` from your project's root directory, `cd` back into it:
> 
> ```bash
> cd ../
> ```
> 
> Now in your dApp's backend (or standalone script), load the *delegate.wasm* file into memory and deploy it with the arguments "amount", "delegator", and "validator" included.

Reviewer: I highly recommend not cloning casper-node in a project's root directory.
Wherever it was cloned, it is ok, but let's not point that out. Please change these lines to say:
```suggestion
Now, navigate back to your project's root directory. In your dApp's backend (or standalone script), load the *delegate.wasm* file into memory and deploy it with the arguments "amount", "delegator", and "validator" included.
```
